rfd/0180-sso-as-mfa-method.md

```diff
@@ -72,171 +194,197 @@ If browser window does not open automatically, open it by clicking on the link:
-```
-
-Note: the MFA prompt is moved to the end to help it stick out over the SSO link.
-
-Stretch: Rather than having the user add an MFA device with additional steps,
-they should be guided through MFA registration and allowed to complete their
-request afterwards. This would also apply without the SSO MFA option.
+##### Adding first MFA Device
 
-```console
-> tsh ssh server01
-MFA is required to access Node "server01"
-Complete an SSO auth check in your local web browser. Open it by clicking on the link:
- http://127.0.0.1:60433/f5858c78-75e1-4f3f-b2c5-69e8e76c0ff9
-Or register an MFA device to complete MFA checks with a tap.
-Choose device type [WEBAUTHN]:
+Usually, a new SSO user without a registered MFA device can add their first
+device without an additional authentication check. With SSO MFA, we can include
+an MFA check even for the first device.
 
-### User can type to continue the registration process or proceed with SSO re-auth.
-```
+TODO:: show tsh add and delete
 
 #### Existing SSO user with registered MFA method
 
 > I have logged into this cluster before with SSO
 > I have registered one or more MFA devices
 > I want to connect to a resource protected by per-session MFA
 
-**New MFA prompts**
+##### SSO MFA prompt
 
 The user will be given the option to pass MFA checks with a registered device
-or with SSO.
+or with SSO, depending on the MFA mode of the user's auth connector.
 
 ```console
 > tsh ssh server01
 MFA is required to access Node "server01"
 Complete an auth check in your local web browser. Open it by clicking on the link:
  http://127.0.0.1:60433/f5858c78-75e1-4f3f-b2c5-69e8e76c0ff9
+ 
 Or tap any security key
 ```
 
-If the user re-authenticates with security key, the web browser should be closed
-automatically.
-
-If SSO is the required MFA method, the security key prompt would be skipped and
-the browser opened automatically.
-
-```console
-> tsh ssh server01
-Re-authentication is required to access this node.
-If browser window does not open automatically, open it by clicking on the link:
- http://127.0.0.1:60433/f5858c78-75e1-4f3f-b2c5-69e8e76c0ff9
-```
-
-#### Teleport Connect
-
-The Teleport connect flow will be identical to `tsh`.
-
-#### WebUI
+Note that the exact prompt may differ depending on which [config scenario](#tsh)
+we are in.
 
-In the WebUI, opening a new tab for MFA automatically would be awkward UX.
-Instead, we should redirect the user in the current tab to complete
-authentication with the IdP. Once complete, the user will be redirected back
-to the original page requiring MFA.
+##### Deleting last MFA device
 
-We will update the current MFA modal to have 3 buttons; `Webauthn`, `SSO`, and
-`Cancel`. Either Webauthn or SSO will be highlighted depending on the cluster 
-configuration.
+Similar to how we can add an SSO MFA check for a [user adding their first device](#adding-first-mfa-device),
+we can use an SSO MFA check for users removing their last device, which is
```

Comment:
Just to clarify, the SSO device counts as "MFA" but not passwordless, correct?

Reply:
I see that this is possible, I think we should allow removing passwordless devices as well. WDYT?

Reply:
I'd be concerned whether the SSO device really is providing a multi-factor check that is equivalent to passwordless. I would err by making it MFA-only to begin with, but I can be convinced otherwise.

Reply:
Right, it's not really passwordless, and I don't think it should be usable for passwordless purposes. But I also think an SSO user should be permitted to delete their last passwordless device in general so long as they have another MFA device. It's not like the user won't be able to login without their passwordless device, which I'm guessing is the pitfall we're trying to avoid for local users. They can just SSO login.

WDYT? Is there something I'm missing about passwordless devices?

Reply:
True, the main worry is the user locking themselves out. Special-casing for UserTypeSSO seems OK, although technically has nothing to do with this RFD. (Feel free to mail me a PR.)

For the present RFD I think we agree that SSOMFADevices are always "MFA", then?